### PR TITLE
Dev 1001 Disable HTML Formatting in Emails

### PR DESCRIPTION
**Pull request by ahsinsiddique**

### CODE REVIEW

File: src/app/store/datatypes.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export class UserState {
   id: number | null;
   whiteList: WhiteList[];
   blackList: BlackList[];
-  settings: Settings;
+  settings: Settings = new Settings();
```
Comment (techevanglist) [inline src/app/store/datatypes.ts:90]: remove it from here and update the `initialState` in store. `users.reducers.ts`.
That is where all the state is being initialized.

File: src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts [hunk 2]
```diff
@@ -342,18 +366,18 @@ export class ComposeMailComponent implements OnInit, AfterViewInit, OnDestroy {
...
 
-    if (this.forwardAttachmentsMessageId) {
-      if (this.editor) {
-        this.updateEmail();
-      } else {
-        setTimeout(() => {
-          this.updateEmail();
-        }, 1000);
-      }
+  toggleHtmleditor(value: boolean) {
```
Comment (techevanglist) [inline src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts:371]: rename `toggleHtmleditor`  to `toggleHtmlEditor`

Comment (techevanglist) [reply src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts:371]: why the function name is toggle if you are passing the value?
the function should rather say `setHtmlEditor`

Comment (ahsinsiddique) [reply src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts:371]: yes, right let me update, thanks

File: src/app/store/datatypes.ts (hunk 1 above)
Comment (ahsinsiddique) [reply src/app/store/datatypes.ts:90]: Okay,

File: src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts [hunk 3]
```diff
@@ -326,12 +328,34 @@ export class ComposeMailComponent implements OnInit, AfterViewInit, OnDestroy {
...
+      }
     } else {
       let content = this.mailData.content ? this.mailData.content : '';
+      if (this.editor) {
+        this.content = this.editor.nativeElement.firstChild.innerHTML;
+      }
+      this.content = this.content && this.content === '' ? content : this.content;
+
       if (this.content) {
-        content += this.getPlainText(this.content);
```
Comment (techevanglist) [inline src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts:334]: why did you remove `+`

Comment (ahsinsiddique) [reply src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts:334]: during testing it was creating issues, let me see again

Comment (techevanglist) [reply src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts:334]: test all cases with html enabled/disabled, send email, forward and reply cases for both html enabled and disabled.

Comment (ahsinsiddique) [reply src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts:334]: sure
